### PR TITLE
Nahuatl Speech Translation

### PR DESCRIPTION
**Pull request by ftshijt**
This is the initial commit for Nahuatl speech translation corpus.

The speech translation manifest files are updated at http://openslr.org/92

### CODE REVIEW

**Comment by ftshijt** [comment]
The project is generally case-free, so I directly convert to lower case during data preparation.

File: egs/puebla_nahuatl/st1/run.sh [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+            --config ${decode_config} \
+            --ngpu 0 \
+            --backend ${backend} \
+            --batchsize 0 \
+            --trans-json ${feat_trans_dir}/split${nj}utt/data_${bpemode}${nbpe}.JOB.json \
+            --result-label ${expdir}/${decode_dir}/data.JOB.json \
+            --model ${expdir}/results/${trans_model}
+
+        score_bleu.sh --case ${tgt_case} --bpe ${nbpe} --bpemodel ${bpemodel}.model \
+            --remove_nonverbal ${remove_nonverbal} \
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:262]: This is not necessary.

File: egs/puebla_nahuatl/st1/run.sh [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    fbankdir=fbank
+    # Generate the fbank features; by default 80-dimensional fbanks with pitch on each frame
+    for x in dev test; do
+        steps/make_fbank_pitch.sh --cmd "$train_cmd" --nj 32 --write_utt2num_frames true \
+            data/${x}_${annotation_id} exp/make_fbank/${x}_${annotation_id} ${fbankdir}
+    done
+
+    # speed perturbation
+    speed_perturb.sh --cmd "${train_cmd}" --cases "tc" --langs "${src_lang} es" data/train_${annotation_id} data/train_${annotation_id}_sp ${fbankdir}
+    # Divide into source and target languages
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:111]: Need a space.

File: egs/puebla_nahuatl/st1/run.sh [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+n_average=5
+use_valbest_average=true
+
+metric=bleu
+
+asr_model=
+mt_model=
+
+# postprocessing related
+remove_nonverbal=true  # remove non-verbal labels such as "( Applaus )"
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:53]: This is not necessary.

File: egs/puebla_nahuatl/st1/run.sh [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+tgt_case=tc
+
+# feature configuration
+do_delta=false
+
+preprocess_config=conf/specaug.yaml
+train_config=conf/train.yaml
+decode_config=conf/decode.yaml
+# train_config=conf/tuning/train_rnn.yaml
+# decode_config=conf/tuning/decode_rnn.yaml
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:41]: Can you remove this?

File: egs/puebla_nahuatl/st1/local/score_bleu.sh [hunk 5]
```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/local/score_bleu.sh:1]: This file is not necessary. This is a scoring script specific to fisher-callhome.

File: egs/puebla_nahuatl/st1/local/normalize_trans.sh [hunk 6]
```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/local/normalize_trans.sh:1]: This file is not necessary.

File: egs/puebla_nahuatl/st1/local/normalize_punctuation.pl [hunk 7]
```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/perl
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/local/normalize_punctuation.pl:1]: This file is not necessary.

Comment (ftshijt) [reply egs/puebla_nahuatl/st1/local/normalize_punctuation.pl]: Thanks! Please pardon me for my mis-check.

File: egs/puebla_nahuatl/st1/run.sh [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+    for x in ${recog_set}; do
+    (
+        decode_dir=decode_${x}_$(basename ${decode_config%.*})
+        feat_trans_dir=${dumpdir}/${x}/delta${do_delta}
+
+        # reset log for RTF calculation
+        if [ -f ${expdir}/${decode_dir}/log/decode.1.log ]; then
+            rm ${expdir}/${decode_dir}/log/decode.*.log
+        fi
+        nj=16
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:243]: Can you move this to general configuration (top of this file)? We may want to make this option according to the hardware.

File: egs/puebla_nahuatl/st1/run.sh [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+echo "dictionary: ${dict}"
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
+    ### Task dependent. You have to check non-linguistic symbols used in the corpus.
+    echo "stage 2: Dictionary and Json Data Preparation"
+    mkdir -p data/lang_1spm/
+
+    # echo "make a non-linguistic symbol list for all languages"
+    # grep sp1.0 data/${train_set}.*/text.${tgt_case} | cut -f 2- -d' ' | grep -o -P '&[^;]*;'| sort | uniq > ${nlsyms}
+    echo "" > ${nlsyms}
+    # cat ${nlsyms}
```
Comment (hirofumi0810) [inline egs/puebla_nahuatl/st1/run.sh:137]:
```suggestion
    cat ${nlsyms}
```

**Review by hirofumi0810 — APPROVED** [review]

**Comment by hirofumi0810** [comment]
@ftshijt LGTM. Thank you for your quick modification!

**Pull request merged**